Source/WebKit/UIProcess/XR/openxr/OpenXRInputSource.cpp

```diff
@@ -0,0 +1,403 @@
+            return false;
+        }
+    };
+
+    for (const auto& profile : openXRInteractionProfiles) {
+        CHECK_XRCMD(createBinding(profile.path, m_gripAction, makeString(m_subactionPathName, s_inputGripPath), bindings));
+        CHECK_XRCMD(createBinding(profile.path, m_pointerAction, makeString(m_subactionPathName, s_inputAimPath), bindings));
+
+#if defined(XR_EXT_hand_interaction)
+        if (OpenXRExtensions::singleton().isExtensionSupported(XR_EXT_HAND_INTERACTION_EXTENSION_NAME)) {
+            RETURN_RESULT_IF_FAILED(createBinding(profile.path, m_pinchPoseAction, makeString(m_subactionPathName, s_inputPinchPath), bindings));
+            RETURN_RESULT_IF_FAILED(createBinding(profile.path, m_pokePoseAction, makeString(m_subactionPathName, s_inputPokePath), bindings));
+        }
+#endif
+
+        for (const auto& button : profile.buttons) {
+            if (!isBindingForHand(button.hand))
+                continue;
+
+            const auto& actions = m_buttonActions.get(button.type);
+            if (button.flags & OpenXRButtonFlags::Click) {
+                ASSERT(actions.press != XR_NULL_HANDLE);
+                CHECK_XRCMD(createBinding(profile.path, actions.press, makeString(m_subactionPathName, button.path, s_pathActionClick), bindings));
+            }
+            if (button.flags & OpenXRButtonFlags::Touch) {
+                ASSERT(actions.touch != XR_NULL_HANDLE);
+                CHECK_XRCMD(createBinding(profile.path, actions.touch, makeString(m_subactionPathName, button.path, s_pathActionTouch), bindings));
+            }
+            if (button.flags & OpenXRButtonFlags::Value) {
+                ASSERT(actions.value != XR_NULL_HANDLE);
+                CHECK_XRCMD(createBinding(profile.path, actions.value, makeString(m_subactionPathName, button.path, s_pathActionValue), bindings));
+            }
+        }
+
+        for (const auto& axis : profile.axes) {
+            auto action = m_axisActions.get(axis.type);
+            ASSERT(action != XR_NULL_HANDLE);
+            CHECK_XRCMD(createBinding(profile.path, action, makeString(m_subactionPathName, unsafeSpan(axis.path)), bindings));
+        }
+    }
+
+    return XR_SUCCESS;
+}
+
+std::optional<PlatformXR::FrameData::InputSource> OpenXRInputSource::collectInputSource(XrSpace localSpace, const XrFrameState& frameState) const
+{
+    PlatformXR::FrameData::InputSource data;
+    data.handedness = m_handedness;
+    data.handle = m_handle;
+    data.targetRayMode = PlatformXR::XRTargetRayMode::TrackedPointer;
+    data.profiles = m_profiles;
+
+    getPose(m_pointerSpace, localSpace, frameState, data.pointerOrigin);
+    PlatformXR::FrameData::InputSourcePose gripPose;
+    if (XR_SUCCEEDED(getPose(m_gripSpace, localSpace, frameState, gripPose)))
+        data.gripOrigin = gripPose;
+
+    Vector<std::optional<PlatformXR::FrameData::InputSourceButton>, openXRButtonTypes.size()> buttons;
+    for (auto& type : openXRButtonTypes) {
```

Comment:
```suggestion
    for (auto type : openXRButtonTypes) {
```

For consistency, it's iterated by value in the other occasion, and it's not being modified